polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -930,23 +934,10 @@ pub mod pallet {
 		/// - `fee_asset_item`: The index into `assets` of the item which should be used to pay
 		///   fees.
 		#[pallet::call_index(1)]
-		#[pallet::weight({
-			let maybe_assets: Result<Assets, ()> = (*assets.clone()).try_into();
-			let maybe_dest: Result<Location, ()> = (*dest.clone()).try_into();
-			match (maybe_assets, maybe_dest) {
-				(Ok(assets), Ok(dest)) => {
-					use sp_std::vec;
-					let count = assets.len() as u32;
-					let mut message = Xcm(vec![
-						WithdrawAsset(assets),
-						SetFeesMode { jit_withdraw: true },
-						InitiateTeleport { assets: Wild(AllCounted(count)), dest, xcm: Xcm(vec![]) },
-					]);
-					T::Weigher::weight(&mut message).map_or(Weight::MAX, |w| T::WeightInfo::teleport_assets().saturating_add(w))
-				}
-				_ => Weight::MAX,
-			}
-		})]
+		#[allow(deprecated)]
+		#[deprecated(
```

Comment:
I'd also add the deprecation notice to the doc comment, since that's sent as metadata to UIs.